### PR TITLE
1924126: Fix profile upload on AWS systems

### PR DESCRIPTION
**Pull request by jirihnidek**
* BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1924126
* Card ID: ENT-3583
* Fix AWS RHUI repo url, when data for profile are gathered.
  Replace REGION with value provided in AWS metadata

### CODE REVIEW

**Comment by wottop** [comment]
Would like to see the result of the CI tests.

**Comment by jirihnidek** [comment]
How to test this?
1) Create scratch build
    `$ tito build --srpm --test`
    `$ brew build --scratch --arch-override=x86_64 rhel-8.5.0-candidate /path/to/sub-man.src.rpm`
2) Install generated rpms on VM runnong on AWS
4) Register system against production candlepin or stage candlepin
3) run: `dnf uploadprofile`

**Comment by cnsnyder** [comment]
@jirihnidek Is this still in progress? I'll be glad to review/test but don't want to do it before it's ready. If it is ready, please take off the "in progress" label, I'll check back early tomorrow :)

**Comment by jirihnidek** [comment]
@cnsnyder Sorry for confusing you. It is not in progress. I only forgot to remove this WIP label.

File: src/cloud_what/providers/aws.py [hunk 1]
```diff
@@ -24,13 +24,19 @@
...
+from typing import Union, Any
 
 from cloud_what._base_provider import BaseCloudProvider
 
 
 log = logging.getLogger(__name__)
 
+# Instance from one region will be redirected to another region's CDS for content
+REDIRECT_MAP = {
+    'us-gov-west-1': 'us-west-2',
```
Comment (cnsnyder) [inline src/cloud_what/providers/aws.py:36]: How static/dynamic is this mapping? Will we have to update this over time?

File: src/cloud_what/providers/aws.py [hunk 2]
```diff
@@ -133,6 +139,39 @@ def is_likely_running_on_cloud(self) -> float:
...
+
+    @staticmethod
+    def rhui_repos(base: Any) -> Any:
+        """
+        Generator of RHUI repositories. Could be used e.g. in for loop
+        :param base: DNF base
+        :return: Yields RHUI repositories
+        """
+        for repo_name, repo in base.repos.items():
+            # TODO: we need more reliable mechanism for detection of RHUI repository, because
```
Comment (cnsnyder) [inline src/cloud_what/providers/aws.py:170]: Is this captured somewhere besides this comment?

File: src/cloud_what/providers/aws.py (hunk 1 above)
Comment (jirihnidek) [reply src/cloud_what/providers/aws.py:36]: It is quite static. I will tell you more details on IRC.

Comment (cnsnyder) [reply src/cloud_what/providers/aws.py:36]: Great! Thank you!

File: src/cloud_what/providers/aws.py (hunk 2 above)
Comment (jirihnidek) [reply src/cloud_what/providers/aws.py:170]: We can create card for this task.

**Pull request merged**